### PR TITLE
TST: refactored tests to use parametrize instead of "for" loop

### PR DESCRIPTION
**Pull request by ShaharNaveh**
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them

### CODE REVIEW

File: pandas/tests/generic/test_series.py [hunk 1]
```diff
@@ -52,33 +52,42 @@ def test_nonzero_single_element(self):
         s = Series([False])
         assert not s.bool()
 
+    @pytest.mark.parametrize(
```
Comment (jreback) [inline pandas/tests/generic/test_series.py:55]: can you just put the Series *iniside* the test functions. (for all of these cases)

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/generic/test_series.py (hunk 1 above)
Comment (ShaharNaveh) [reply pandas/tests/generic/test_series.py:55]: I'm not sure I understand, what does it means "inside" the test functions?
I would really appreciate if you can post of example code for what you ment:)

Comment (jreback) [reply pandas/tests/generic/test_series.py:55]: the parameters should be like `[np.nan, pd.NatT]` then simply construct what you need in the function

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @ShaharNaveh

**Pull request merged**